### PR TITLE
[Bugfix:InstructorUI] Add user modal does not clear on close

### LINKED ISSUE #4991: Hitting close on add user modal does not clear modal
**Describe the bug**
When opening the modal to add a user, use auto-complete the find a user and fill out all fields, hit close, and then hit the button to open the modal, the modal is still filled totally in with the previous user's information.

**Expected behavior**
The second time I open the modal, all fields are blank.

**To Reproduce**
1. Open new student modal
2. Type in "pearsr" and then tab to auto-complete all fields
3. Hit close on modal
3. Hit button to re-open modal

**Configuration**
 - OS: [e.g. Windows, Linux, ...] macOS
 - Browser [e.g. chrome, safari] Chrome, Safari

**Additional context**
If we are afraid of losing data by accident when hitting the close button, we should add a mechanism to warn a user then that hitting close will lose the data, and not have it be that they can just re-open the modal and keep working on that student.

### PR DESCRIPTION
**Pull request by mukul-kmr-jha**
### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)

### What is the current behavior?
The form modal opened by the `New Student` button keeps the data filled by the auto-completion feature even after hitting `Close` button 
Similarly, the `New Grader` form modal is storing the previous user's information.

### What is the new behavior?
The state of form is being reset on close event. Closes #4991

### CODE REVIEW

File: site/public/js/userform.js [hunk 1]
```diff
@@ -204,9 +204,11 @@ function autoCompleteOnUserId(user_information) {
 }
 
 function completeUserFormInformation(user) {
-    var form = $("#edit-user-form");
+  var form = $("#edit-user-form");
+
+  $('[name="edit_user"]', form).val("true");
 
-    $('[name="user_numeric_id"]', form).val(user['user_numeric_id']);
+  $('[name="user_numeric_id"]', form).val(user['user_numeric_id']);
```
Comment (elihschiff) [inline site/public/js/userform.js]: It looks like you removed some white-space here and that messes with the indentation. I think it looks better with the original white-space where everything lines up.

File: site/public/js/userform.js [hunk 2]
```diff
@@ -204,9 +204,11 @@ function autoCompleteOnUserId(user_information) {
 }
 
 function completeUserFormInformation(user) {
-    var form = $("#edit-user-form");
+  var form = $("#edit-user-form");
```
Comment (elihschiff) [inline site/public/js/userform.js]: It looks like you removed some white-space here and that messes with the indentation. I think it looks better with the original white-space where everything lines up.
This also will require fixing the new line 209 to be at the same indentation level.

File: site/public/js/userform.js [hunk 3]
```diff
@@ -51,7 +51,6 @@ function newUserForm() {
     $("#edit-student-modal-title").css('display','none');
     $("#edit-grader-modal-title").css('display','none');
     $("#user-form-already-exists-error-message").css('display','none');
-    $('[name="edit_user"]', form).val("false");
```
Comment (elihschiff) [inline site/public/js/userform.js]: I do not think you want to remove this line. With your code, if you edit a user, press cancel, then add a new user. The edit_user value will be true. 
Lines 208/209 might also relate to this issue.

File: site/public/js/userform.js [hunk 4]
```diff
@@ -282,11 +305,12 @@ function clearValidityWarnings() {
 }
 
 function closeButton() {
-    $('#edit-user-form').css('display', 'none');
-    if($('[name="edit_user"]').val() == 'true') {
+    // If form contain some data, prompt the user for data loss
+    if(!isUserFormEmpty() && confirm('Changes to the Form will be lost!')){
```
Comment (MasterOdin) [inline site/public/js/userform.js]: `Form` should lowercased to `form`

File: site/public/js/userform.js [hunk 5]
```diff
@@ -282,11 +305,12 @@ function clearValidityWarnings() {
...
         $('[name="user_id"]', $("#edit-user-form")).val("");
         clearUserFormInformation();
     }
+    $('#edit-user-form').css('display', 'none');
```
Comment (MasterOdin) [inline site/public/js/userform.js]: I think we want that this should only happen if we confirm. Right now, if we hit cancel after we get the confirm box, the modal will still close, but the data will remain the next time we open the modal.

**Comment by MasterOdin** [comment]
Close to merging this (thanks for the patience on this), just one last thing.

File: site/public/js/userform.js (hunk 5 above)
Comment (mukul-kmr-jha) [reply site/public/js/userform.js]: Added changes and rebased.
PTAL, Thanks!

**Comment by MasterOdin** [comment]
Thanks for your work on this!

**Pull request merged**